src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -256,13 +256,35 @@ def datetime_to_index(
             )
         )
 
+    def index_to_datetime(self, index: int) -> datetime | None:
+        """Convert the given index to the underlying datetime.
+
+        Index 0 corresponds to the oldest timestamp in the buffer. The
+        resulting datetime can be outside the range of the buffer.
+
+        Args:
+            index: Index to convert.
+
+        Returns:
+            Datetime index where the value for the given index can be found.
+            Or None if the buffer is empty.
```

Comment:
Return docs needs an extra indentation if it is more than one line, because of some weird feature that you can document different types as different "return items" (like args):

```suggestion
            The index as a `datetime` where the value for the given index can be found.
                Or `None` if the buffer is empty.
```

I think an alternative could be:

```
        Returns:
            datetime: The index as a `datetime` where the value for the given index can be found.
            None: If the buffer is empty.
```
But we never used it in the SDK. It might be interesting to see how it renders, maybe is not a bad idea.

In any case, I suggest having a look at the rendered docs (`pip install -e .[dev-mkdocs] && mkdocs serve`) to make sure it is rendering properly.

Reply:
I just checked this, and the ring buffer is private, so it is not really rendered. I also see that this is existing code. Anyway, just for the sake of knowning what's the best approach to use, here is how the `__getitem__` in the `MovingWindow` renders:

```
        Returns:
            A float if the key is a number or a timestamp.
            an numpy array if the key is a slice.
```

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/17b7df23-6ccc-4830-8563-8b1c60672a0b)

The alternative (splitting the types) doesn't render extremely well either it basically renders as a list of arguments:

```
        Returns:
            float: The individual value at `key`.
            ArrayLike: The sub-window matching the bounds in `key` `slice`.
```

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/d9c93594-2157-4aab-bb76-ff7e2ec53946)

Here is the simple (fixed) version with only one `Returns:` *item*:

```
        Returns:
            A float if the key is a number or a timestamp.
                an numpy array if the key is a slice.
```

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/537e4e5f-d5a8-44df-8425-cedeedb6323e)

I think we should go with that for now.